Training/Test/UnitTest1.cs

```diff
@@ -1,8 +1,47 @@
+using ClassLibrary;
 namespace Test {
    [TestClass]
    public class UnitTest1 {
+      TDoubleEndedQueue<int> mNumbers = new ();
+
       [TestMethod]
-      public void TestMethod1 () {
+      public void TestEnqueueAtFront () {
+         for (int i = 0; i < 4; i++) mNumbers.EnqueueFront (i + 1);
+         for (int j = 0; j < 4; j++) Assert.AreEqual (j + 1, mNumbers.DequeueRear ());
+      }
+
+      [TestMethod]
+      public void TestEnqueueAtRear () {
+         for (int i = 0; i < 4; i++) mNumbers.EnqueueRear (i + 1);
+         for (int j = 0; j < 4; j++) Assert.AreEqual (j + 1, mNumbers.DequeueFront ());
+      }
+
+      [TestMethod]
+      public void TestRearDequeue () {
+         for (int i = 0; i < 5; i++) {
+            if ((i + 1) > 2) mNumbers.EnqueueFront (i + 1);
+            else mNumbers.EnqueueRear (2 - i);
+         }
+         for (int j = 0; j < 5; j++) Assert.AreEqual (j + 1, mNumbers.DequeueRear ());

```

Comment:
Can you explain what you are testing here, and what the expected queue state will be after the first for loop ?

Reply:
Rear deque sir. I am doing dequeue at the end and checking if they are in ascending order from 1. It will check if the elements dequeue from end are 1,2,3,4,5